scene/gui/popup.cpp

```diff
@@ -233,7 +233,7 @@ Size2 PopupPanel::_get_contents_minimum_size() const {
 
 void PopupPanel::_update_child_rects() {
 	Vector2 cpos(theme_cache.panel_style->get_offset());
-	Vector2 csize(get_size() - theme_cache.panel_style->get_minimum_size());
+	Vector2 csize; // There must always be `panel`, so safe to set csize below.
```

Comment:
Is the assumption guaranteed, that "panel" is the first Control-child?
In cases, where "panel" is not the first child, the loop below can lead to unexpected results, since the size of Control nodes can be set, without the initialization of ´csize`.

The user can change the order of child nodes and can interfere with this assumption.
I believe, that it would be more robust, to initialize `csize` before the for-loop.

Reply:
Well-reasoned. I made the assumption that "panel" would always remain as the first child, not considering that the user could change the order. Is it safe to assume "panel" cannot be removed altogether?

Reply:
Users removing internal nodes ... sounds like a recipe for disaster.
While these cases should not be expected, if a crash can be prevented with a simple `!= null` check, I believe, that it should be included.

Reply:
`panel` is an internal child, so user has no access to it unless they want to purposefully break the node. It can't be done by accident.

Reply:
I made the suggested change to set `csize` correctly, regardless of `panel` child index. Per KoBeWi and that I believe it is standard practice to not protect internal nodes, I did not add a `null` or `is_instance_valid` check. There are many warnings in the docs about bad consequences of deleting internal nodes, but those are usually exposed.